_inc/lib/components.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Components Library
+ *
+ * Load and display a pre-rendered component
+ */
+class Jetpack_Components {
+	/**
+	 * Load and display a pre-rendered component
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return string The component markup
+	 */
+	public static function render_component( $name, $props ) {
+		wp_enqueue_style( 'wp-components' );
+
+		$rtl = is_rtl() ? '.rtl' : '';
+		wp_enqueue_style( 'jetpack-components', plugins_url( "_inc/blocks/components{$rtl}.css", JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
```

Comment:
Just a minor comment that `jetpack-components` handle and `blocks/components` directory feels confusing, considering that Jetpack contains "DOPS components" in `_inc/client/components`. Should the handle be `jetpack-blocks-components` instead?

Reply:
See https://github.com/Automattic/jetpack/blob/7f96a8cf205f04a82624d046aa2e2a6684045816/_inc/lib/admin-pages/class.jetpack-admin-page.php#L205-L206

Reply:
> Just a minor comment that `jetpack-components` handle and `blocks/components` directory feels confusing, considering that Jetpack contains "DOPS components" in `_inc/client/components`. Should the handle be `jetpack-blocks-components` instead?

To clarify, it's a filename (`components{$rtl}.css`) in the `_inc/blocks/` directory. The latter happens to be the build directory for our Gutenberg blocks (and plugins).

The filename OTOH comes from the [entry point setting](https://github.com/Automattic/jetpack/pull/13070/files#diff-09575ea3a128e617b9c541a2b7262da5R82) for the webpack config for `StaticSiteGeneratorPlugin`. I've chosen this in turn to correspond to the `extensions/shared/components/index.jsx` path (and I hope to move the `upgrade-nudge/` dir into `extensions/shared/components/` in a follow-up, making that dir the central place for components needed by multiple Gutenberg blocks).

Within the context of Gutenberg blocks, `components` (with no other prefix and/or suffix) makes sense, so I'd like to keep it for (source) directories and filenames. However, we can change the entry point name (which in turn determines the built CSS filename). `blocks-components` could work there, to distinguish from `dops-components`, but it doesn't sound great. I'd rather not include the `jetpack-` prefix with anything that clearly is inside of Jetpack; the `wp_enqueue_style` handle is a different matter though. I'd be all for calling it `jetpack-blocks-components` -- if it wasn't for precedent like

> See
> 
> https://github.com/Automattic/jetpack/blob/7f96a8cf205f04a82624d046aa2e2a6684045816/_inc/lib/admin-pages/class.jetpack-admin-page.php#L205-L206

:roll_eyes: 

---

I think a lot of this question boils down to the (current) split between JP dashboard vs Gutenberg extensions, as reflected by the two separate webpack configs. The more present this split is on someone's radar, the more obvious is the distinction between these two different sets of components.

Reply:
>blocks-components could work there, to distinguish from dops-components, but it doesn't sound great.

I agree the `blocks-` prefix isn't great and clarity around naming is especially tricky because of two separate build pipelines for this stuff. :-/